### PR TITLE
Phase 2: Training & Distillation Quality

### PR DESCRIPTION
**Pull request by Super-Genius**
## Summary

**Phase 2: Training & Distillation Quality**
**Goal:** Knowledge distillation converges with configurable KD loss, temperature sweeping produces analysis data, specialist training produces valid LoRA adapters, evaluation persists metrics per run, and a rules-based router maps query patterns to specialists.
**Status:** Verified (8/8 UAT, 70 tests passed)

### Plan 01: Config Extensions
Extended pipeline.yaml with training convergence, eval_gates, and router config sections. Per-specialist distill_loss_target (medical=2.0, code/qa=2.5, encycl/patents=3.0), val_loss_threshold, and eval_gates. ConfigLoader extended with 3 new validators. transitions 0.9.3 installed.
**Files:** config/pipeline.yaml, config/loader.py, config/specialists/*.yaml (5 files)

### Plan 02: Convergence Tracking & Distillation Quality
ConvergenceTracker with plateau-based early stopping and two-tier thresholds (warning + hard-stop). SweepAnalyzer produces structured JSON sweep analysis. Distiller extended with convergence tracking per temperature.
**Files:** distill/convergence.py, distill/sweep_analyzer.py, distill/distillation.py

### Plan 03: Training Quality Validation
Multi-prong AdapterValidator: loadability (safetensors/npz files, non-zero tensor norms), validation loss (below per-specialist threshold), behavioral diff (>5% token difference). Inert adapter detection via all-zero weight check.
**Files:** training/adapter_validator.py

### Plan 04: Evaluation Metrics & Gating
MetricStore with versioned JSON persistence and trend analysis. ThresholdAdapter with LLM-based threshold recommendations and auto-apply gate (confidence > 0.9, change < 20%). SpecialistEvaluator extended with evaluate_and_persist(). Benchmarker extended with gate_check() for auto-gating.
**Files:** eval/metric_store.py, eval/threshold_adapter.py, eval/evaluator.py, eval/benchmarker.py

### Plan 05: Rules-Based Router
Greenfield router module with YAML-driven keyword/regex/token-density classification. GQHSM-compatible JSON state machine wrapping transitions library. Fallback chaining mirrors Phase 1 teacher cascade pattern.
**Files:** router/__init__.py, router/plan.py, router/rules.py, router/state_machine.py, config/router_rules.yaml

## Requirements Addressed

DIST-01: KD Loss Convergence with plateau-based early stopping
DIST-02: Temperature Sweeping with structured JSON analysis
DIST-03: Synthetic Data Quality filtering
TRAIN-01: LoRA Adapter Validity with multi-prong checks
TRAIN-02: Hyperparameter Configurability via per-specialist YAML
TRAIN-03: Evaluation Metrics with versioned persistence
ROUTE-01: Rule-Based Query Classification (keyword/regex/token-density)
ROUTE-02: Specialist Selection and Execution Plan with fallback chaining

## Verification

- [x] UAT: 8/8 tests passed (70 total)
- [x] Config: 12/12 self-tests
- [x] ConvergenceTracker: 6/6 behavior tests
- [x] SweepAnalyzer: 7/7 tests
- [x] AdapterValidator: 7/7 tests
- [x] MetricStore: 21/21 tests
- [x] Router: 17/17 tests
- [x] GQHSM schema: 7 states, 9 transitions

### CODE REVIEW

File: gnus-poc/router/state_machine.py [hunk 1]
```diff
@@ -0,0 +1,508 @@
...
+        "name": "router_classifier",
+        "version": 1,
+        "description": "Rules-based query classification for specialist routing",
+        "namespace": "gnus.router",
+    },
+    "states": [
+        {
+            "name": "pre_classify",
+            "isStartState": True,
+            "entryAction": "log_entry",
```
[the remark this answers is not in the document]
Comment (Super-Genius) [reply gnus-poc/router/state_machine.py:398]: @codex please recheck this with the updated code

File: gnus-poc/eval/metric_store.py [hunk 2]
```diff
@@ -55,6 +84,137 @@ def __init__(self, project_root: Optional[Path] = None):
...
+
+        Args:
+            niche: Specialist niche name (e.g. ``"medical"``).
+
+        Returns:
+            List of prior-run dicts, earliest first.  Empty if no prior
+            runs exist or all files were corrupt.
+        """
+        prior: List[Dict[str, Any]] = []
+        pattern = f"{niche}_*.json"
```
[the remark this answers is not in the document]
Comment (Super-Genius) [reply gnus-poc/eval/metric_store.py:141]: Fixed in 6f6b566 — `load_prior()` now excludes stems ending in `_sgfp4_metrics` so SGFP4 artifacts are no longer loaded as prior eval runs. Regression test: `test_load_prior_excludes_sgfp4_artifacts`.

**Comment by Super-Genius** [comment]
@codex rescan from new commits
